velox/functions/lib/Re2Functions.cpp

```diff
@@ -376,9 +377,41 @@ class Re2SearchAndExtract final : public VectorFunction {
 
 class LikeConstantPattern final : public VectorFunction {
  public:
+  void determinePatternKind() {
+    if (!likeutils::isWildcardInPattern(pattern_)) {
+      patternKind_ = PatternKind::fixedPattern;
+    } else if ((prefixLength_ = likeutils::isPrefixPattern(pattern_))) {
+      patternKind_ = PatternKind::prefixPattern;
+    } else if ((suffixLength_ = likeutils::isSuffixPattern(pattern_))) {
+      patternKind_ = PatternKind::suffixPattern;
```

Comment:
If we end up here, we would have scanned the pattern string 3 times. It would be more efficient to determine the pattern kind with a single pass other the pattern string.

Reply:
Thanks for the input, simplified these functions into a single function determinePatternKind to avoid scanning the pattern multiple times.